CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -510,6 +510,12 @@ function PlayerHospital:onEndDay()
     end
   end
 
+  for _, staff in ipairs(self.staff) do
+    if staff:isIdle() then
+      self.world.dispatcher:answerCall(staff)
+    end
+  end
```

Comment:
On reinspecting some code, we do use idle a fair bit for some actions as a wait (I found vaccinate and multi-use object). Would these likely get misinterpreted by this code?

Reply:
Just judging by the question, it seems to me that `isIdle` is ambiguous. So one step is to decide yes or no on your question and then document it with `isIdle`, possibly also improving the name of the function to better match its use.

Also, you may want to split its use between 2 or more functions with different names, so it becomes easier to get the right function. To assist the programmer, it's likely a good idea to mention the related functions in lua doc and/or put them close together, to improve the chance the right function gets found.

As for the functionality, in the calls dispatcher you added `not staff:getRoom()` as condition, which apparently means "a wandering staff member". While I'd prefer an explicit function for that notion, eg `hasNoWork` or so, that check may possibly be good here?

Reply:
If you add a new function for the latter, please also add a link to that in the lua doc of `getRoom` to avoid further spreading of the `getRoom` function for this purpose.

Another measure that can be done is to make the `getRoom` private, although that has much larger consequences (it should be called with `self` only), not sure if that is feasible.

Reply:
`isIdle` is actually quite a hefty function to determine a true idle state now that I got round to reading it. It is definitely much more than the mere `IdleAction` I thought it was. It's documented inside nicely enough in the function, not for doc though.